TFLite/src/common/Storage.py

```diff
@@ -0,0 +1,208 @@
+
+                    # Adding Node attributes and their values to 
+                    # columns and values
+                    attrs = vars(node)
+                    for item in attrs.items():
+                        if item[0] != 'label' and item[0] != 'value':
+                            columns.append(item[0])
+                            values.append(item[1])
+
+                    columns = tuple(columns)
+                    values = tuple(values)
+
+                    # Insert
+                    batch.insert(
+                        table = 'Operators',
+                        columns = columns,
+                        values=  [values]
+                    )
+                    operator_id += 1
+
+    # Internal function to load tensor(Edge) data into Tensors table
+    def _load_tensors(self, graph):
+
+        if len(graph.edges) == 0:
+            return
+
+        # Dictionary to store source nodes and destination nodes of edges
+        to_nodes = dict()
+        from_nodes = dict()
+
+        # Populating to_nodes and from_nodes using BFS
+        adj_list = graph.adj_list
+        start_node_indices = graph.start_node_indices
+
+        # Queue and visit status of nodes for BFS
+        queue = Queue()
+        vis = [False] * len(graph.nodes)
+
+        for start_node_index in start_node_indices:
+            if not vis[start_node_index]:
+                vis[start_node_index] = True
+                queue.put(start_node_index)
+
+                # BFS
+                while not queue.empty():
+                    src_node_index = queue.get()
+
+                    if src_node_index not in adj_list:
+                        continue
+
+                    for [edge_index, dest_node_index] in adj_list[src_node_index]:
+
+                        if edge_index not in from_nodes:
+                            from_nodes.update({edge_index : set()})
+                        
+                        from_nodes[edge_index].add(src_node_index + 1)
+
+                        if edge_index not in to_nodes:
+                            to_nodes.update({edge_index : set()})
+
+                        to_nodes[edge_index].add(dest_node_index + 1)
+
+                        if not vis[dest_node_index]:
+                            vis[dest_node_index] = True
+                            queue.put(dest_node_index)
+
+        # Surrogate Id for tensors
+        tensor_id = 0
+
+        # Number of edges to be processed per batch
+        num_attributes = len(vars(graph.edges[0]))
+        num_edges_per_batch = (20000 + num_attributes - 1) // num_attributes
+        
+        edge_indices = list(to_nodes.keys())
+        num_edges = len(edge_indices)
+
+        # Inserting data into the Tensors table
+        # Looping till all edges are inserted
+        while tensor_id < num_edges:
+            # Creating and inserting a batch
+            with self.database.batch() as batch:
```

Comment:
Perhaps for future, it might be worth adding some retry logic here based on backoff in case of insertion failures due to deadline exceeded while doing sequence of batch insertions.
Please feel free to add a TODO and leave it as it is.

Reply:
Added a TODO in subsequent PR